server/etcdserver/txn/txn.go

```diff
@@ -93,20 +93,28 @@ func put(ctx context.Context, txnWrite mvcc.TxnWrite, p *pb.PutRequest) (resp *p
 	return resp, nil
 }
 
-func DeleteRange(kv mvcc.KV, dr *pb.DeleteRangeRequest) (*pb.DeleteRangeResponse, error) {
-	txnWrite := kv.Write(traceutil.TODO())
+func DeleteRange(ctx context.Context, lg *zap.Logger, kv mvcc.KV, dr *pb.DeleteRangeRequest) (resp *pb.DeleteRangeResponse, trace *traceutil.Trace, err error) {
+	trace = traceutil.Get(ctx)
+	// create put tracing if the trace in context is empty
+	if trace.IsEmpty() {
```

Comment:
This looks like more than just unifying the args, should we do this as a second commit?

Reply:
This is unifying args, I'm adding ctx to args and trace to return as all function have.

Reply:
Thanks for clarification, understand now.

Nit - Comment above references put but I think it should say delete?

Reply:
Done